wpengine/faustjs
# Migrate WordPress block asset files from PHP to JSON format

## Tasks


## Description

These changes aim to fix the failing unzipping action reported in #1633.

Traditional WordPress blocks generate *.asset.php files. However, many hosting environments restrict PHP files in the uploads directory. This change proposes converting these files to index.asset.json, aligning with hosting constraints and improving compatibility.

Supporting tests have been added to both the cli package and WordPress plugin.

![blockset-architecture](https://github.com/wpengine/faustjs/assets/6676674/d84f2f09-616c-477d-8556-da9ea9e8dfec)

## Related Issue(s):

#1633

## Testing

1. Checkout this branch: `fix/blockset-issue-1633`
2. Update the block support example project (`examples/next/block-support/)` to be a WordPress site hosted on WP Engine.
3. Add the updated [faustwp.zip](https://github.com/wpengine/faustjs/files/13704494/faustwp.zip) to your WordPress site.
4. Run `npm run blockset -w @faustwp/block-support-example` from the root of this cloned repo.
5. Observe "WordPress: Blockset sync complete" in your terminal.

    <img width="904" alt="Screenshot 2023-11-17 at 11 11 29 AM" src="https://github.com/wpengine/faustjs/assets/6676674/4eada937-fbe3-464a-ac16-f9bd4ebb540b">
5. Observe the blocks defined in your Next.js app have been compiled and are present in your connected WordPress site's uploads directory.

    <img width="1217" alt="Screenshot 2023-11-17 at 11 10 19 AM" src="https://github.com/wpengine/faustjs/assets/6676674/fcc5cdb7-8dfd-404b-b59e-6f8743a7fffe">
